# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/targets/angle_common/moz.build
# Commit: 3a91885d47e0
# Full Hash: 3a91885d47e0ef7fba9749562d52300eea67c651
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-11-11 21:52:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 77cee77c0307 -r 3a91885d47e0 gfx/angle/targets/angle_common/moz.build
--- a/gfx/angle/targets/angle_common/moz.build	Fri Nov 08 16:41:42 2019 +0000
+++ b/gfx/angle/targets/angle_common/moz.build	Fri Nov 08 17:06:00 2019 +0000
@@ -2,32 +2,26 @@
 
 include('../../moz.build.common')
 
-DEFINES['ANGLE_ENABLE_DEBUG_ANNOTATIONS'] = True
 #DEFINES['ANGLE_IS_64_BIT_CPU'] = True
 #DEFINES['CERT_CHAIN_PARA_HAS_EXTRA_FIELDS'] = True
-#DEFINES['CHROMIUM_BUILD'] = True
-#DEFINES['COMPONENT_BUILD'] = True
-DEFINES['DYNAMIC_ANNOTATIONS_ENABLED'] = '1'
-#DEFINES['FULL_SAFE_BROWSING'] = True
+DEFINES['CR_CLANG_REVISION'] = '"357692-1"'
+DEFINES['DYNAMIC_ANNOTATIONS_ENABLED'] = '0'
+#DEFINES['NDEBUG'] = True
 DEFINES['NOMINMAX'] = True
-#DEFINES['NO_TCMALLOC'] = True
-DEFINES['NTDDI_VERSION'] = '0x0A000003'
+DEFINES['NTDDI_VERSION'] = 'NTDDI_WIN10_RS2'
+#DEFINES['NVALGRIND'] = True
 #DEFINES['PSAPI_VERSION'] = '2'
-#DEFINES['SAFE_BROWSING_CSD'] = True
-#DEFINES['SAFE_BROWSING_DB_LOCAL'] = True
 DEFINES['UNICODE'] = True
 #DEFINES['USE_AURA'] = '1'
 #DEFINES['WIN32'] = True
 #DEFINES['WIN32_LEAN_AND_MEAN'] = True
 #DEFINES['WINAPI_FAMILY'] = 'WINAPI_FAMILY_DESKTOP_APP'
 DEFINES['WINVER'] = '0x0A00'
-#DEFINES['WTF_USE_DYNAMIC_ANNOTATIONS'] = '1'
 DEFINES['_ATL_NO_OPENGL'] = True
 DEFINES['_CRT_RAND_S'] = True
 DEFINES['_CRT_SECURE_NO_DEPRECATE'] = True
-#DEFINES['_DEBUG'] = True
 DEFINES['_HAS_EXCEPTIONS'] = '0'
-#DEFINES['_HAS_ITERATOR_DEBUGGING'] = '0'
+#DEFINES['_HAS_NODISCARD'] = True
 DEFINES['_SCL_SECURE_NO_DEPRECATE'] = True
 DEFINES['_SECURE_ATL'] = True
 DEFINES['_UNICODE'] = True
@@ -44,24 +38,56 @@
 ]
 
 #CXXFLAGS += [
+#    '-D__DATE__=',
+#    '-D__TIME__=',
+#    '-D__TIMESTAMP__=',
+#    '-fcolor-diagnostics',
+#    '-fcomplete-member-pointers',
+#    '-fcrash-diagnostics-dir=../tools/clang/crashreports',
+#    '-fdebug-compilation-dir',
+#    '-fmerge-all-constants',
+#    '-fmsc-version=1916',
+#    '-fno-standalone-debug',
+#    '-gcodeview-ghash',
+#    '-instcombine-lower-dbg-declare=0',
+#    '-m64',
+#    '-mllvm',
+#    '-no-canonical-prefixes',
+#    '-Wextra-semi',
+#    '-Wheader-hygiene',
+#    '-Wimplicit-fallthrough',
+#    '-Wno-builtin-macro-redefined',
+#    '-Wno-c++11-narrowing',
+#    '-Wno-ignored-pragma-optimize',
+#    '-Wno-missing-field-initializers',
+#    '-Wno-nonportable-include-path',
+#    '-Wno-undefined-var-template',
+#    '-Wno-unneeded-internal-declaration',
+#    '-Wno-unused-parameter',
+#    '-Wnon-virtual-dtor',
+#    '-Wstring-conversion',
+#    '-Wtautological-overlap-compare',
+#    '-Wthread-safety',
+#    '-Wunneeded-internal-declaration',
+#    '-Xclang',
+#    '.',
 #    '/bigobj',
-#    '/d2FastFail',
-#    '/D__DATE__=',
-#    '/D__TIME__=',
-#    '/D__TIMESTAMP__=',
+#    '/Brepro',
 #    '/FS',
-#    '/GF',
 #    '/GR-',
+#    '/guard:cf,nochecks',
+#    '/Gw',
 #    '/Gy',
-#    '/MDd',
-#    '/Ob0',
-#    '/Od',
+#    '/MT',
+#    '/O1',
+#    '/Ob2',
+#    '/Oi',
+#    '/Oy-',
 #    '/TP',
 #    '/utf-8',
 #    '/W4',
 #    '/wd4091',
 #    '/wd4100',
-#    '/wd4117',
 #    '/wd4121',
 #    '/wd4127',
 #    '/wd4200',
@@ -103,17 +129,23 @@
 #    '/we4244',
 #    '/we4456',
 #    '/we4458',
+#    '/we4715',
 #    '/we4800',
 #    '/we4838',
 #    '/WX',
+#    '/X',
+#    '/Z7',
+#    '/Zc:inline',
 #    '/Zc:sizedDealloc-',
-#    '/Zi',
+#    '/Zc:twoPhase',
 #]
 
 SOURCES += [
     '../../checkout/src/common/aligned_memory.cpp',
+    '../../checkout/src/common/android_util.cpp',
     '../../checkout/src/common/angleutils.cpp',
     '../../checkout/src/common/debug.cpp',
+    '../../checkout/src/common/event_tracer.cpp',
     '../../checkout/src/common/Float16ToFloat32.cpp',
     '../../checkout/src/common/mathutil.cpp',
     '../../checkout/src/common/matrix_utils.cpp',
@@ -153,8 +185,17 @@
     ]
 
 #LDFLAGS += [
-#    '/DEBUG',
+#    '--color-diagnostics',
+#    '/DEBUG:GHASH',
+#    '/FIXED:NO',
+#    '/INCREMENTAL:NO',
+#    '/OPT:ICF',
+#    '/OPT:NOLLDTAILMERGE',
+#    '/OPT:REF',
 #    '/pdbaltpath:%_PDB%',
+#    '/PDBSourcePath:C:/Code/angle/out',
+#    '/PROFILE',
+#    '/TIMESTAMP:1564894800',
 #    '/WX',
 #]
 